File: app/src/main/java/ua/in/quireg/anothermovieapp/movieDetails/ReviewsRecyclerViewAdapter.java
package ua.in.quireg.anothermovieapp.movieDetails;

import android.content.Context;
import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import ua.in.quireg.anothermovieapp.R;
import ua.in.quireg.anothermovieapp.common.CursorRecyclerViewAdapter;
import ua.in.quireg.anothermovieapp.common.MovieReview;

public class ReviewsRecyclerViewAdapter extends CursorRecyclerViewAdapter<RecyclerView.ViewHolder> {
    private static final String LOG_TAG = ReviewsRecyclerViewAdapter.class.getSimpleName();

    public ReviewsRecyclerViewAdapter() {
        super(null);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, Cursor cursor) {
        MovieReviewViewHolder movieHolder = (MovieReviewViewHolder) viewHolder;

        MovieReview movie_review = MovieReview.fromCursor(cursor);
        movieHolder.reviewAuthor.setText(movie_review.author);
        movieHolder.reviewContent.setText(movie_review.content);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        Context appContext = parent.getContext();
        View view = LayoutInflater.from(appContext)
                .inflate(R.layout.fragment_movie_review_item, parent, false);
        return new MovieReviewViewHolder(view);
    }

    public class MovieReviewViewHolder extends RecyclerView.ViewHolder {
        TextView reviewAuthor;
        TextView reviewContent;

        MovieReviewViewHolder(View view) {
            super(view);
            reviewAuthor = view.findViewById(R.id.review_author);
            reviewContent = view.findViewById(R.id.review_content);
        }

        @NonNull
        @Override
        public String toString() {
            return super.toString() + " '" + reviewAuthor.getText() + "'";
        }
    }
}

File: app/src/main/java/com/anothermovieapp/movieDetails/ReviewsRecyclerViewAdapter.kt
package com.anothermovieapp.movieDetails

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.anothermovieapp.R
import com.anothermovieapp.repository.EntityDBMovieReview

class ReviewsRecyclerViewAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    protected var innerData = mutableListOf<EntityDBMovieReview>()

    fun setData(newData: List<EntityDBMovieReview>?) {
        newData?.let {
            val prevSize = innerData.size
            val newSize = newData.size
            innerData.clear()
            innerData.addAll(newData)
            notifyItemRangeInserted(prevSize, newSize)
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val appContext = parent.context
        val view = LayoutInflater.from(appContext)
                .inflate(R.layout.fragment_movie_review_item, parent, false)
        return MovieReviewViewHolder(view)
    }

    inner class MovieReviewViewHolder internal constructor(view: View) : RecyclerView.ViewHolder(view) {
        var reviewAuthor: TextView
        var reviewContent: TextView
        override fun toString(): String {
            return super.toString() + " '" + reviewAuthor.text + "'"
        }

        init {
            reviewAuthor = view.findViewById(R.id.review_author)
            reviewContent = view.findViewById(R.id.review_content)
        }
    }

    companion object {
        private val LOG_TAG = ReviewsRecyclerViewAdapter::class.java.simpleName
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val movieHolder = holder as MovieReviewViewHolder
        val movie_review: EntityDBMovieReview = innerData[position]
        movieHolder.reviewAuthor.text = movie_review.author
        movieHolder.reviewContent.text = movie_review.content
    }

    override fun getItemCount(): Int {
        return innerData.size
    }

}